bots/heroes/hunter_main.lua

```diff
@@ -0,0 +1,427 @@
+			end
+		elseif itemNullBlade:CanActivate() then
+			if unitSelf:HasState(State_Hunter_Ability1_Buff) then -- Verifies there is a debuff on BH
+				bActionTaken = core.OrderItemEntityClamp(botBrain, unitSelf, itemNullBlade, unitSelf) -- Purges the silence debuff
+			end
+		elseif nInRange and (nLastHarassUtility > botBrain.nManaBurn1Threshold) then --Enemy is in range and nullfire passes threshold check
+				bActionTaken = core.OrderItemEntityClamp(botBrain, unitSelf, itemNullBlade, unitTarget) --Casts Nullfire blade on enemy.
+		end
+	end
+	
+	if not bActionTaken then
+		return object.harassExecuteOld(botBrain)
+	end
+end
+object.harassExecuteOld = behaviorLib.HarassHeroBehavior["Execute"]
+behaviorLib.HarassHeroBehavior["Execute"] = HarassHeroExecuteOverride
+
+--------------------------
+--Feast Function Pieces --
+--------------------------
+
+local function feastingActionsUtility(botBrain)
+
+	local abilFeast = skills.abilFeast	
+	local unitSelf = core.unitSelf 	
+	
+	--If blood hunter is severely wounded, or has access mana and is mildly wounded...
+	if abilFeast:CanActivate() and ((unitSelf:GetHealthPercent() < .35) or (unitSelf:GetManaPercent() > .75  and unitSelf:GetHealthPercent() < .85)) and unitSelf:IsAlive() then
+		return 25 --Run feastingActionsExecute
+	end
+	
+	return 0
+end
+
+local function feastingActionsExecute(botBrain)
+
+	local unitSelf = core.unitSelf
+	local unitFeastTarget = nil --Setting up target for feasting
+	local unitECreep = core.localUnits["EnemyCreeps"]
+	
+	local abilFeast = skills.abilFeast	
+	
+	local bActionTaken = false
+	
+	if unitECreep then --If there are local enemy creeps,
+	
+		local nBestTargetHealth = unitSelf:GetHealth()/5
+	
+		for _, unitTarget in pairs(unitECreep) do --looks for the highest health creep to feast on!
+			local targetHealth = unitTarget:GetMaxHealth() 
+			if targetHealth > nBestTargetHealth and unitTarget:IsAlive() then 
+				unitFeastTarget = unitTarget
+				nBestTargetHealth = targetHealth
+			end
+		end
+		
+		-- Our good ol' "27 B-Stroke 6"
+		if unitFeastTarget then --If the unit is a feast target has been chosen
+		
+			local vecMyPosition = unitSelf:GetPosition() --BH's position
+			local vecTargetPosition = unitFeastTarget:GetPosition() --FeastTarget's position
+			local nTargetDistanceSq = Vector3.Distance2DSq(vecMyPosition, vecTargetPosition) --Distance between BH and FeastTarget
+			local nRange = abilFeast:GetRange()
+			
+			if nTargetDistanceSq < (nRange * nRange) then --If the target creep is in range |Error = function arguments expected 
+			
+				bActionTaken = core.OrderAbilityEntity(botBrain, abilFeast, unitFeastTarget, false) --consume the best creep
+				
+			elseif nTargetDistanceSq > (nRange * nRange) then --If the creep is not in range, get in range
+			
+				bActionTaken = core.OrderMoveToPosClamp(botBrain, unitSelf, vecTargetPosition, false) --Command to move towards the targeted creep
+				
+			end
+		end
+	end
+	
+	return false
```

Comment:
heh bActionTaken

Do you want to make a bot description or do I make one? See https://github.com/honteam/Heroes-of-Newerth-Bots/blob/Community/stringtables/interface_en.str#L10596